x-pack/plugins/inference/server/chat_complete/errors.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { InferenceTaskError, type UnvalidatedToolCall } from '@kbn/inference-common';
+import { i18n } from '@kbn/i18n';
+import {
+  ChatCompletionErrorCode,
+  ChatCompletionTokenLimitReachedError,
+  ChatCompletionToolNotFoundError,
+  ChatCompletionToolValidationError,
+} from '@kbn/inference-common/src/chat_complete/errors';
+
+export function createTokenLimitReachedError(
+  tokenLimit?: number,
+  tokenCount?: number
+): ChatCompletionTokenLimitReachedError {
+  return new InferenceTaskError(
+    ChatCompletionErrorCode.TokenLimitReachedError,
+    i18n.translate('xpack.inference.chatCompletionError.tokenLimitReachedError', {
+      defaultMessage: `Token limit reached. Token limit is {tokenLimit}, but the current conversation has {tokenCount} tokens.`,
+      values: { tokenLimit, tokenCount },
+    }),
+    { tokenLimit, tokenCount }
+  );
+}
+
+export function createToolNotFoundError(name: string): ChatCompletionToolNotFoundError {
+  return new InferenceTaskError(
+    ChatCompletionErrorCode.ToolNotFoundError,
+    `Tool ${name} called but was not available`,
```

Comment:
Does this need `i18n`?

Reply:
Same as https://github.com/elastic/kibana/pull/193464#discussion_r1823163921 - I'd say no (fwiw this was only moved, not introduced by the PR)